src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -180,6 +181,35 @@ public interface OnHitEffect
 		}
 	};
 	
+	/**
+	 * A knockback effect that does a right-vector (relative to the player's orientation) and direction to enemy dot calculation (relative to the player's orientation) to spread enemies horizontally via knockback.
+	 */
+	OnHitEffect SPREADING_KNOCKBACK = (stack, target, attacker) -> {
+		
+		//Attacker's right direction and position
+		double pRightSin = Math.sin(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0);
+		double pRightCos = Math.cos(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0);
+		double pX = attacker.getX(), pZ = attacker.getZ();
+		//Other entity position
+		double eX = target.getX(), eZ = target.getZ();
+		
+		//Direction vector player -> entity
+		Vec2 dirToEntity = new Vec2((float) (eX - pX), (float) (eZ - pZ));
+		dirToEntity = dirToEntity.normalized();
+		
+		//Dot product of direction and player's right
+		double dot = (pRightSin * dirToEntity.x) + (pRightCos * dirToEntity.y);
+		double expDot = 1.5 * Math.pow(1 - (Math.abs(dot)), 0.3);
```

Comment:
If I understand the equation for `expDot` correctly, it's meant to make the sideways knockback stronger for entities directly in front of the player. Is that right?

Reply:
Exactly. It adds a force based on the relative position, the addition is weaker when closer to the player's right, that is, it becomes closer to default knockback.